plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -442,6 +442,16 @@ def _onSlicingFinishedMessage(self, message):
         self.backendStateChange.emit(BackendState.Done)
         self.processingProgress.emit(1.0)
 
+        for line in self._scene.gcode_list:
+            replaced = line.replace("{print_time}", str(Application.getInstance().getPrintInformation().currentPrintTime.getDisplayString(DurationFormat.Format.ISO8601)))
+            replaced = replaced.replace("{filament_amount}", str(Application.getInstance().getPrintInformation().materialLengths))
+            replaced = replaced.replace("{filament_weight}", str(Application.getInstance().getPrintInformation().materialWeights))
+            # TODO: calculate filament cost
+            replaced = replaced.replace("{filament_cost}", "Not yet implemented")
```

Comment:
str(Application.getInstance().getPrintInformation().materialCosts)

Seems to work :)